flow/core/kernel/scenario/traci.py

```diff
@@ -0,0 +1,885 @@
+        x = makexml('configuration',
+                    'http://sumo.dlr.de/xsd/netconvertConfiguration.xsd')
+        t = E('input')
+        t.append(E('node-files', value=self.nodfn))
+        t.append(E('edge-files', value=self.edgfn))
+        if types is not None:
+            t.append(E('type-files', value=self.typfn))
+        if connections is not None:
+            t.append(E('connection-files', value=self.confn))
+        x.append(t)
+        t = E('output')
+        t.append(E('output-file', value=self.netfn))
+        x.append(t)
+        t = E('processing')
+        t.append(E('no-internal-links', value='%s' % no_internal_links))
+        t.append(E('no-turnarounds', value='true'))
+        x.append(t)
+        printxml(x, self.net_path + self.cfgfn)
+
+        subprocess.call(
+            [
+                'netconvert -c ' + self.net_path + self.cfgfn +
+                ' --output-file=' + self.cfg_path + self.netfn +
+                ' --no-internal-links="%s"' % no_internal_links
+            ],
+            shell=True)
+
+        # collect data from the generated network configuration file
+        error = None
+        for _ in range(RETRIES_ON_ERROR):
+            try:
+                edges_dict, conn_dict = self._import_edges_from_net()
+                return edges_dict, conn_dict
+            except Exception as e:
+                print('Error during start: {}'.format(e))
+                print('Retrying in {} seconds...'.format(WAIT_ON_ERROR))
+                time.sleep(WAIT_ON_ERROR)
+        raise error
+
+    def generate_net_from_osm(self, net_params):
+        """Generate .net.xml files from OpenStreetMap files.
+
+        This is accomplished by calling the sumo ``netconvert`` binary. Only
+        vehicle roads are included from the networks.
+
+        Parameters
+        ----------
+        net_params : flow.core.params.NetParams type
+            network-specific parameters. Different networks require different
+            net_params; see the separate sub-classes for more information.
+
+        Returns
+        -------
+        edges : dict <dict>
+            Key = name of the edge
+            Elements = length, lanes, speed
+        connection_data : dict < dict < list<tup> > >
+            Key = name of the arriving edge
+                Key = lane index
+                Element = list of edge/lane pairs that a vehicle can traverse
+                from the arriving edge/lane pairs
+        """
+        # specify the location of the input osm file
+        osm_path = net_params.osm_path
+
+        # specify the location of the output file
+        netfn = "%s.net.xml" % self.name
+
+        # generate the network file with sumo
+        net_cmd = "netconvert --osm-files {0} --output-file {1}".\
+            format(osm_path, self.cfg_path + netfn)
+
+        # this handles removing all roads in the network that cannot be ridden
+        # by vehicles
```

Comment:
We should put an issue up about this, presumably at some point we may want to have pedestrians

Reply:
Maybe, but this will be more clear when we get there